donkeycar/templates/train.py

```diff
@@ -67,6 +68,7 @@ def __getitem__(self, index):
         throttles = []
 
         is_inferred = type(self.keras_model) is KerasInferred
+        is_categorical = type(self.keras_model) is KerasCategorical
```

Comment:
Actually I don't like this `if/else` stuff here, I would prefer to move this data processing that is specific to the `KerasPilot` into a virtual method there. Then each model could decide how to massage the data by itself, that would be much more OO. @tikurahul - let me know if you agree.

Reply:
Yes, in general the  model is coupled with the input format. So that makes a lot of sense to me. 

However, lets try and figure out a good API first - because as we get to the more exotic models, I have a feeling that we might have to do a lot more.

Reply:
Sure, we can keep this for now, and I'll make a PR with a design proposal.